keps/3589-manage-jobs-selectively/README.md

```diff
@@ -0,0 +1,232 @@
+    - [Story 1](#story-1)
+- [Design Details](#design-details)
+  - [API](#api)
+  - [Implementation](#implementation)
+  - [Test Plan](#test-plan)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [Do nothing](#do-nothing)
+  - [Validating Admission Policies](#validating-admission-policies)
+<!-- /toc -->
+
+## Summary
+
+This KEP adds an optional field to the Kueue `Configuration` struct that enables
+the effect of `manageJobsWithoutQueueName` to be controlled on a per-namespace
+basis for all Kueue integrations.
+
+-->
+
+## Motivation
+
+When Kueue is deployed on multi-tenant clusters, it is essential that cluster
+admins be able to configure Kueue such that users cannot bypass its quota system.
+In particular, there must be a robust and recommended mechanism for configuring Kueue
+such that all Pod-creating Kinds in user namespaces are subject to management by Kueue.
+
+Currently, the primary mechanism for doing this is setting `manageJobsWithoutQueueName` to
+true.  When `manageJobsWithoutQueueName` is true, then Kueue will manage all instances of
+Kinds with Kueue integrations, even if they do not have a `kueue.x-k8s.io/queue-name` label.
+As a practical matter, Pods running in system namespaces must be exempted from being
+managed by Kueue even if `manageJobsWithoutQueueName` is true.  Therefore Kueue's Pod
+integration limits the scope of `manageJobsWithoutQueueName` by restricting its effect to
+Pods that are in namespaces that match against the `integrations.podOptions.namespaceSelector`.
+
+The recently added `Deployment` and `StatefulSet` integrations are built on top of the
+`Pod` integration and therefore the effect of `manageJobsWithoutQueueName` for these
+two integrations is also filtered by `integrations.podOptions.namespaceSelector`.  A similar
+rationale holds for `Deployment` and less strongly for `StatefulSet` that some mechanism
+to limit `manageJobsWithoutQueueName` is required otherwise it would be impossible in
+practice to ever set `manageJobsWithoutQueueName` to true.
+
+No other integrations beyond these three consider the `namespaceSelector` when deciding whether or not to
+suspend jobs that do not have a `queue-name` label.  This results in an irregular API and makes it
+harder to configure Kueue such that quotas will be enforced (cannot be bypassed by users).
+The irregularity between `batchv1/Job` and `Deployment` is likely to be especially confusing
+to users and cluster admins who would view both of these types as "built-in" to Kubernetes and
+would expect them to be configured in the same way.
+
+### Goals
+
+Provide a simple and uniform API to enable cluster admins to configure Kueue
+such that quotas will be enforced for specific namespaces.
+
+### Non-Goals
+
+Consider any scope for filtering `manageJobsWithoutQueueName` that is not based on namespaces.
```

Comment:
This is just comment.

My primary concern for this KEP was this one.
I would not like to accept this feature in any case since this tries to re-implement the Gatekeeper, VAP, and MAP.

Reply:
Yes, I fully agree we don't want this mechanism to be more complex, and +1 to reject features not based on namespaces, but for namespaces we already have the complication in `podOptions.namespaceSelector`, and making it uniformly supported seems like good idea when we have Deployments and StatefulSets supported.

I agree with the direction to use VAP and MAP, but this will probably take many releases as MAP is still alpha in 1.32, and we need to propose details for VAP and MAP generation to avoid issues with hierarchical jobs (as described in the KEP).

Reply:
Thanks for expanding that. If anyone wants to know VAP and MAP introduction details, please refer https://github.com/kubernetes-sigs/kueue/issues/3636 and https://github.com/kubernetes-sigs/kueue/issues/3637.